### ISSUE TITLE
Add confusion_matrix method to TabularPredictor

### ISSUE THREAD

**Issue by celestinoxp** [awslabs/autogluon#5341]
## Description
Add a new method to TabularPredictor to compute and optionally plot the confusion matrix for classification tasks.
Includes basic unit tests and supports normalized output, display, and saving plots.

## References
[scikit-learn ](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html)
